ui-v2/src/hooks/task-run-concurrency-limits.ts

```diff
@@ -0,0 +1,205 @@
+import type { components } from "@/api/prefect";
+import { getQueryService } from "@/api/service";
+import {
+	queryOptions,
+	useMutation,
+	useQueryClient,
+	useSuspenseQuery,
+} from "@tanstack/react-query";
+
+export type TaskRunConcurrencyLimit = components["schemas"]["ConcurrencyLimit"];
+export type TaskRunConcurrencyLimitsFilter =
+	components["schemas"]["Body_read_concurrency_limits_concurrency_limits_filter_post"];
+
+/**
+ * ```
+ *  🏗️ Task run concurrency limits queries construction 👷
+ *  all   =>   ['task-run-concurrency-limits'] // key to match ['task-run-concurrency-limits', ...
+ *  list  =>   ['task-run-concurrency-limits', 'list'] // key to match ['task-run-concurrency-limits', 'list', ...
+ *             ['task-run-concurrency-limits', 'list', { ...filter1 }]
+ *             ['task-run-concurrency-limits', 'list', { ...filter2 }]
+ *  details => ['task-run-concurrency-limits', 'details'] // key to match ['task-run-concurrency-limits', 'details', ...
+ *             ['task-run-concurrency-limits', 'details', id1]
+ *             ['task-run-concurrency-limits', 'details', id2]
+ * ```
+ * */
+export const queryKeyFactory = {
+	all: () => ["task-run-concurrency-limits"] as const,
+	lists: () => [...queryKeyFactory.all(), "list"] as const,
+	list: (filter: TaskRunConcurrencyLimitsFilter) =>
+		[...queryKeyFactory.lists(), filter] as const,
+	details: () => [...queryKeyFactory.all(), "details"] as const,
+	detail: (id: string) => [...queryKeyFactory.details(), id] as const,
+};
+
+// ----- 🔑 Queries 🗄️
+// ----------------------------
+export const buildListTaskRunConcurrencyLimitsQuery = (
+	filter: TaskRunConcurrencyLimitsFilter = { offset: 0 },
+) =>
+	queryOptions({
+		queryKey: queryKeyFactory.list(filter),
+		queryFn: async () => {
+			const res = await getQueryService().POST("/concurrency_limits/filter", {
+				body: filter,
+			});
+			return res.data ?? [];
+		},
+	});
```

Comment:
We probably want some polling on this query to ensure the user sees relatively up-to-date info on how many slots are occupied while staying on this page. Somewhere between 30 and 60 seconds seems reasonable to me.

Reply:
all, let me update that